cql3/statements/select_statement.cc

```diff
@@ -1651,10 +1651,16 @@ void select_statement::check_needs_filtering(const restrictions::statement_restr
 {
     // non-key-range non-indexed queries cannot involve filtering underneath
     if (!_parameters->allow_filtering() && (restrictions.is_key_range() || restrictions.uses_secondary_indexing())) {
+        const auto& ck_restrictions = *restrictions.get_clustering_columns_restrictions();
+        const auto& pk_restrictions = *restrictions.get_partition_key_restrictions();
         // We will potentially filter data if either:
         //  - Have more than one IndexExpression
         //  - Have no index expression and the column filter is not the identity
-        if (restrictions.need_filtering()) {
+        if (restrictions.need_filtering()
+            // See #7608.
+            || ((pk_restrictions.empty() || has_token(pk_restrictions.expression))
+                && !ck_restrictions.empty() && !find_atom(ck_restrictions.expression, expr::is_multi_column)
```

Comment:
how does is_multi_column enter here? Do we have a test covering it?

Reply:
I think I added it before #9085 was fixed.  But it's now incorrect, as Cassandra seems to treat it the same as other CK restrictions.

In v2, I dropped the multi-column check and added a cql-pytest case that passes on both Cassandra and Scylla.

Reply:
Done.